classes/client/object_client.php

```
@@ -27,15 +27,34 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-interface object_client {
-    public function __construct($config);
-    public function register_stream_wrapper();
-    public function get_fullpath_from_hash($contenthash);
-    public function get_trash_fullpath_from_hash($contenthash);
-    public function delete_file($fullpath);
-    public function rename_file($currentpath, $destinationpath);
-    public function get_seekable_stream_context();
-    public function get_availability();
-    public function get_maximum_upload_size();
-    public function verify_object($contenthash, $localpath);
+class object_client {
+
+    public function get_availability() {
+        return false;
+    }
+
+    public function register_stream_wrapper() {
+        return false;
+    }
+
+    /**
+     * Does the storage support pre-signed URLs.
+     *
+     * @return bool.
+     */
+    public function support_signed_urls() {
+        return false;
+    }
+
+    /**
+     * Generates pre-signed URL to storage file from its hash.
+     *
+     * @param string $contenthash File content hash.
+     *
+     * @return string.
+     */
+    public function generate_signed_url($contenthash) {
+        return 'Not supported';
```

should we just throw an exception here?